compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp

```diff
@@ -562,7 +550,9 @@ void CPUMaterializeEncodingPass::runOnOperation() {
   if (!materializeEncodingFn) {
     return signalPassFailure();
   }
-  MaterializeEncodingTypeConverter typeConverter(materializeEncodingFn);
+  bool transposeNarrowN = !isVMVXBackend(targetAttr);
```

Comment:
Please add some comments about why VMVX is excluded.

EDIT: do we only want this being applied on llvm-cpu? What's the issue on VMVX side, I thought they will navigate to the same kernel if we apply the change unconditionally. It can make logics easier.

Reply:
Thanks for catching that. I was being lazy. Back when I was desperate to get this to finally work, I cut corners by disabling this whole transposition on VMVX to limit the number of edge cases I needed to debug. Now that everything else is working, I tried reenabling that, ran into a bug, debugged it: I had forgotten a transposition in the materialization of dispatch.tensor.load/store ops, that for some reason only manifested itself on VMVX. Fixed. Now it's all much simpler, that `transposeNarrowN` boolean is gone.